src/shogun/classifier/svm/OnlineSVMSGD.cpp

```diff
@@ -145,7 +145,7 @@ bool OnlineSVMSGD::train(std::shared_ptr<Features> data)
 	float64_t wnorm = linalg::dot(m_w, m_w);
 	io::info("Norm: {:.6f}, Bias: {:.6f}", wnorm, bias);
 
-	return true;
+	return shared_from_this()->as<Machine>();
```

Comment:
```suggestion
	return shared_from_this();
```

Reply:
I think you don't need to write this redundant `->as<Machine>()`.

Reply:
You are right, but I wanted to make it explicit that we're returning a shared pointer of type "Machine". If you insist, I'll change it.

Reply:
```
shogun/src/shogun/classifier/svm/OnlineSVMSGD.cpp:148:25: error: could not convert ‘std::enable_shared_from_this<_Tp>::shared_from_this() [with _Tp = shogun::SGObject]()’ from ‘shared_ptr<shogun::SGObject>’ to ‘shared_ptr<shogun::Machine>’
  return shared_from_this();
```

It seems manual cast is mandatory, checkie this https://stackoverflow.com/questions/4491420/enable-shared-from-this-and-inheritance

Reply:
oh, sorry, my previous thought is wrong.